reframe/core/exceptions.py

```diff
@@ -98,6 +98,12 @@ class TaskDependencyError(ReframeError):
     '''
 
 
+class MaxFailError(ReframeError):
```

Comment:
Just raise a `ReframeError` directly.

Reply:
Then the output of async policy, in the summary of failures, will be simply ```* Reason: aborted due to ReframeError``` instead of ```* Reason: aborted due to MaxFailError``` that it is now.
Should we change more in the output?

Reply:
I don't like that it fails like that. Let me play a bit with this to get the feeling.